cabal-install/Distribution/Client/ProjectBuilding.hs

```diff
@@ -1184,7 +1184,18 @@ buildInplaceUnpackedPackage verbosity
           --TODO: [required eventually] this doesn't track file
           --non-existence, so we could fail to rebuild if someone
           --adds a new file which changes behavior.
-          allSrcFiles <- allPackageSourceFiles verbosity scriptOptions srcdir
+          allSrcFiles <-
+            let trySdist    = allPackageSourceFiles verbosity scriptOptions srcdir
+                -- This is just a hack, to get semi-reasonable file
+                -- listings for the monitor
+                tryFallback = filter (not . ("dist" `isPrefixOf`))
+                                <$> getDirectoryContentsRecursive srcdir
+            in if elabSetupScriptCliVersion pkg >= Version [1,17] []
+                  then do r <- trySdist
+                          if null r
+                            then tryFallback
```

Comment:
btw, why do we want this even though Cabal >= 1.17 at this point?

maybe we should emit a warning that we had to fallback to enumerate files ourselves even though Cabal >= 1.17 was used